src/test.js

```diff
@@ -342,6 +346,11 @@ Test.prototype = {
 		config.current = undefined;
 
 		function logSuiteEnd( module ) {
+
+			// Reset `module.hooks` to ensure that anything referenced in these hooks
+			// has been released to be garbage collected.
+			module.hooks = {};
```

Comment:
something in the suiteEnd probably was using hooks?

Reply:
This has been a doozy digging into. I'm still somewhat uncertain, but it seems that the `test-on-node` grunt task is leaking tests (not properly waiting for the `QUnit.done` that is setup in `tests/log.js` which does a `setTimeout` _inside_ the done callback).

I've pushed a small cleanup that prevents the failure, but there is definitely some work needed in this area to clean up the leakage between tests.

Reply:
I believe that as of this commit, when QUnit is tries to report the "No tests run" error, it is unable to do so now, because:

```js
  			if (module.hooks[handler].length) {
  			                          ^

TypeError: Cannot read property 'length' of undefined
    at processHooks
```

I ran into this as part of reproducing https://github.com/qunitjs/qunit/issues/1405 locally.